nativelib/src/main/resources/cre2.cpp

```diff
@@ -0,0 +1,305 @@
+/*
+  https://github.com/marcomaggi/cre2 (0.3.0)
+
+  Source  file  for  cre2, a  C   language  wrapper  for RE2:  a   regular
+  expressions library by Google.
+
+  Copyright (c) 2012 Marco Maggi <marco.maggi-ipsu@poste.it>
+  Copyright (c) 2011 Keegan McAllister
+  All rights reserved.
+
+  Redistribution and  use in source and binary  forms, with or
+  without  modification,  are   permitted  provided  that  the
+  following conditions are met:
+
+  1.  Redistributions of  source  code must  retain the  above
+     copyright  notice,  this   list  of  conditions  and  the
+     following disclaimer.
+
+  2. Redistributions  in binary form must  reproduce the above
+     copyright  notice,  this   list  of  conditions  and  the
+     following  disclaimer in  the documentation  and/or other
+     materials provided with the distribution.
+
+  3.  Neither the  name of  the author  nor the  names  of his
+     contributors may  be used to endorse  or promote products
+     derived from this software without specific prior written
+     permission.
+
+  THIS  SOFTWARE  IS PROVIDED  BY  THE  COPYRIGHT HOLDERS  AND
+  CONTRIBUTORS   ``AS  IS''   AND  ANY   EXPRESS   OR  IMPLIED
+  WARRANTIES,  INCLUDING,  BUT  NOT  LIMITED TO,  THE  IMPLIED
+  WARRANTIES OF  MERCHANTABILITY AND FITNESS  FOR A PARTICULAR
+  PURPOSE ARE  DISCLAIMED.  IN NO  EVENT SHALL THE  AUTHORS OR
+  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+  NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+  LOSS  OF USE,  DATA, OR  PROFITS; OR  BUSINESS INTERRUPTION)
+  HOWEVER CAUSED  AND ON ANY  THEORY OF LIABILITY,  WHETHER IN
+  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+  OTHERWISE)  ARISING  IN ANY  WAY  OUT  OF  THE USE  OF  THIS
+  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+Scala-Native modifications
+
+ + cre2_find_named_capturing_groups (https://github.com/marcomaggi/cre2/pull/14)
+
+ - cre2_opt_* (options getter)
+ - cre2_easy_match
+ - cre2_match_*
+ - cre2_strings_to_ranges
+ - cre2_replace & cre2_global_replace (we use functions accepting regex)
+ - cre2_possible_match_range
+ - cre2_check_rewrite_string
+ - cre2_extract & cre2_extract_re
```

Comment:
Imagine writing an instruction what needs to be done to the original file to make this one. It doesn't include the fact that we add `scalanative_` prefix for example.